plugins/cue_points/quiz/services/QuizUserEntryService.php

```diff
@@ -24,20 +24,32 @@ public function submitQuizAction($id)
 			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $id);
 
 		if ($dbUserEntry->getType() != QuizPlugin::getCoreValue('UserEntryType',QuizUserEntryType::QUIZ))
-		{
-			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_TYPE, $dbUserEntry->getType());
-		}
+			throw new KalturaAPIException(KalturaQuizErrors::PROVIDED_ENTRY_IS_NOT_A_QUIZ, $id);
+
 		/**
 		 * @var QuizUserEntry $dbUserEntry
 		 */
 		$score = $dbUserEntry->calculateScore();
 		$dbUserEntry->setScore($score);
-//		$dbUserEntry->setStatus(QuizUserEntryStatus::QUIZ_SUBMITTED);
+		KalturaLog::debug("Quiz score is [" .$score."]");
+    
 		$dbUserEntry->setStatus(QuizPlugin::getCoreValue('UserEntryStatus', QuizUserEntryStatus::QUIZ_SUBMITTED));
+ 		$userEntry = new KalturaQuizUserEntry();
+		$userEntry->fromObject($dbUserEntry, $this->getResponseProfile());
+		$entryId = $dbUserEntry->getEntryId();
+		$entry = entryPeer::retrieveByPK($entryId);
```

Comment:
Please move all validations, both the entry and the quiz to the beginning of the method, before calculating the score or creating new objects.